### ISSUE TITLE
Vim gets very slow, the longer VSCode seems to be open

### ISSUE THREAD

**Issue by RononDex** [VSCodeVim/Vim#3957]
**Describe the bug**
Since a few days / weeks vim has become incredibly slow for me, and it gets worse the longer vscode is open. Any mode change takes around 1s when vscode is open for 1h or so, and around 2s when its open 2h or so.
This also includes using easymotion

**To Reproduce**
1. Keep vscode open for some time (and do some work)
2. vim gets slower and slower as time progresses
3. restart of vscode makes it fast again

**Expected behavior**
Performance does not degrade with time

**Environment (please complete the following information):**

Extension version: 1.9.0
VS Code version: Code 1.36.1 (2213894ea0415ee8c85c5eea0d0ff81ecc191529, 2019-07-08T22:59:35.033Z)
OS version: Windows_NT x64 10.0.16299

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz (12 x 3696)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>oop_rasterization: disabled_off<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.87GB (11.71GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>

**Comment by RononDex** [comment]
UPDATE:
I figured out what causes it to get slow.
If  I disable "vim.statusBarColorControl": false,, it gets fast again

So updating the status bar is what takes so long

**Comment by adriandeveloper** [comment]
I'm still having this issue.  

"vim.statusBarColorControl" isn't enabled for me and it's lagging.

**Comment by renodubois** [comment]
Also getting a similar issue (really bad lag in INSERT mode, doesn't really have any rhyme or reason to it) and I also have `vim.statusBarColorControl` disables.

**Comment by danditomaso** [comment]
I'm having this issue as well, and i dont have ```vim.statusBarColorControl``` enabled. So i'm hoping to see a fix to this soon, primarily while switching from normal to insert mode can take up to a minute switching modes.

**Comment by adriandeveloper** [comment]
> I'm having this issue as well, and i dont have `vim.statusBarColorControl` enabled. So i'm hoping to see a fix to this soon, primarily while switching from normal to insert mode can take up to a minute switching modes.

Yeah. I notice the typing slows down a lot. Its like it's in a bottleneck and I have to wait for Vim to catch up to what I typed.

**Comment by stevenguh** [comment]
I haven't actually had Vim Code Vim slow down over time, but I have it slow down when I open a large file.
If you are seeing slow down on vim operations, maybe try using the profiling tool in the Developer tools and figure out what causes it?

**Comment by sebastianhoitz** [comment]
I'm seeing a similar issue. For me, when I press J or K for like a second the cursor afterwards is still catching up while in normal mode.

This is especially bad if the explorer sidebar is open. If I hide it, the performance gets better.
I have tried disabling the statusBarColorControl, but this did not change anything.

**Comment by joolius** [comment]
Same issue. Currently the plugin is unusable for me.

**Comment by jakechiam** [comment]
Yea, my vim plugin has running really slow, for the past few versions of vscode. 
Curiously though, I have no problems with the vim plugin when using vscode insiders. 

I'm curious if someone else having this issue can try enabling vim in vscode insiders to see if they have the same experience?

**Comment by AlexMenor** [comment]
I was blaming vscode for this but indeed seems like its the Vim plugin.

**Comment by velcroz** [comment]
> I'm seeing a similar issue. For me, when I press J or K for like a second the cursor afterwards is still catching up while in normal mode.

Same for me, I often overshoot my target code by a few lines because of that

**Comment by mrjoedang** [comment]
> > I'm seeing a similar issue. For me, when I press J or K for like a second the cursor afterwards is still catching up while in normal mode.
> 
> Same for me, I often overshoot my target code by a few lines because of that

I am getting the same issue.

**Comment by rodrei** [comment]
same experience here

**Comment by austinsantoso** [comment]
Same issue here, sometimes after a while, it causes any keyboard input to not be registered at all.

**Comment by dvthai89** [comment]
same here, especially on insert mode.

**Comment by alex-cory** [comment]
As a temporary workaround, I noticed when uninstalling it, then reinstalling it, it will work well again. This is temporary, but at least it works.

**Comment by Twiggeh** [comment]
Bumping this, Vim plugin getting unusable

**Comment by noslenac** [comment]
I don't see any performance issues while keeping it open. I have it open all day on two computers - Linux Mint and Windows 10 (build 18363.628).

Some very old Windows 10 builds seem to struggle occasionally with Electron apps such as VS Code.

**Comment by alex-cory** [comment]
I'm on mac Mojave and although I don't experience this issue all the time, it does happen from time to time and it's really annoying. I noticed it happening a lot when I was copying svg xml into JS files.

**Comment by pianocomposer321** [comment]
I'm experiencing the same lag in motion. Honestly, this is one of the biggest things that makes me almost want to go back to just regular old vim!

**Comment by stefabat** [comment]
Also in my case I experience a very annoying lag in motion. I am starting to seriously thinking to change editor just for this.

**Comment by pianocomposer321** [comment]
@stefabat, you might want to try [onivim](https://github.com/onivim/oni). I can't wait until [onivim 2](https://github.com/onivim/oni2) gets out, b/c it's going to support vscode extensions. But final release for that isn't until after June. :(

**Comment by pianocomposer321** [comment]
btw, I think my lag is because of enabling vim.foldFix. However, I really need to have the fix, so disabling it is not an option for me.

**Comment by stefabat** [comment]
@pianocomposer321 onivim and onivim 2 look amazing, I did not know them. However, in the meantime I found an acceptable solution with this vscode extension: [neo-vim](https://marketplace.visualstudio.com/items?itemName=asvetliakov.vscode-neovim).
It is much faster than the vim extension, and has more or less everything I need.

**Comment by pianocomposer321** [comment]
I have tried the extension too (in fact, I uninstalled the vscodevim extension in favor of it), and I agree that performance-wise it is much better. However, it has the same problem that enabling vim.foldFix solved with vscodevim, which is that it opens folds when moving the cursor over them. Several issues have been opened about this, but I don't think there is currently a fix in the works. Incidentally, I believe enabling vim.foldFix is also what caused the lag on vscodevim.

**Comment by ecbrodie** [comment]
Also running into this problem, with serious lag between my typing and the characters appearing on the screen. I am using the latest version of MacOS (10.15.3), VSCode (1.42.0) and the VIM extension (1.12.4). I don't notice any difference with intellisense being triggered or not. I also notice no difference between editing a large file or a small file (I could easily reproduce my issue in a brand new file).

Just to ensure that other extensions were not getting in the way, I booted up VSCode with a fresh copy of the `~/.vscode` directly and only installed the VIM extension. Still getting the issue.

I am sadly forced to disable this extension now because of how much this issue is getting in the way of me being able to write code. I doubt this is something that this extensions maintainers and community desire for :(

**Comment by pianocomposer321** [comment]
Just out of curiosity, how many of you who are experiencing the lag have enabled vim.foldFix?

**Comment by mattydebie** [comment]
I don't have foldFix enabled

**Comment by ecbrodie** [comment]
`vim.foldFix` does not even appear to be a valid extension, according to the list of available settings in my VSCode installation.

**Comment by pianocomposer321** [comment]
It is. see [https://github.com/VSCodeVim/Vim#-faq](https://github.com/VSCodeVim/Vim#-faq). Not sure what you mean about the list of available settings.

**Comment by pianocomposer321** [comment]
For those of you here using vscode-neovim who are experiencing fold opening, try adding `nmap j gj` and `nmap k gk` to your init.vim. Note: this will also make the cursor go to the wrapped part of a line instead of going to the next line if a line is wrapped, but often this is the behavior you want anyway.

**Comment by ecbrodie** [comment]
@pianocomposer321 what I meant by "list of available settings" is based on when I opened up all configuration options available in VSCode preferences. I saw nothing that was called "foldFix".

Yes, I acknowledge what is stated in this project's FAQ, but based on `foldFix`'s description, I doubt that this setting is related to my issue because I rarely, if ever, utilize code folding.

It's sad that there's little indication that this bug will ever get fixed :(

**Comment by pianocomposer321** [comment]
I solved the problem by adding this to my settings.json:

```
"vim.normalModeKeyBindingsNonRecursive": [
        {
			"before": [
				"j"
			],
			"commands": [
				"cursorDown"
			]
		},
		{
			"before": [
				"k"
			],
			"commands": [
				"cursorUp"
			]
		}
],
```

I also commented out `"vim.foldfix": true`, because `cursorUp`/`cursorDown` already skip over folds.

Note: this fix breaks things like `10j` and `10k`, but I rarely use counts with j and k.

**Comment by pianocomposer321** [comment]
Also, I think part of my problem was from the extension "Bracket Pairs Colorizer 2". The redrawing of colors every time the cursor moved made the lag worse. Uninstalling the extension helped, but did not solve the problem completely.

**Comment by osdiab** [comment]
Yep, this is unbearably bad on my machine too. Switched to the aforementioned `neovim` extension and it's a lot smoother (but the editing experience is not quite as good for me yet, gotta tinker). will be giving Oni a try as well!

**Comment by Sheepolution** [comment]
Here's a gif of how slow it is for me.

What happens in the gif:
1. In NORMAL mode I go down and let go of the key when I'm on line 35. What happens is that the editor still needs to catch up and continues to move down until line 64.
2. In INSERT mode I do the same. Going down and releasing the key on line 35. I end up on line 42, most likely because I let go of the key a bit too late.
3. In NORMAL mode I move all the way down. It takes about 3 seconds.
4. In INSERT mode I move all the way down. It takes about 2 seconds.

![vim-slow](https://user-images.githubusercontent.com/2232780/80277843-22aa1180-86f2-11ea-8815-2f4e61d3514f.gif)

This is a VSCode instance with only this file/tab open. Hopefully this can be fixed because the catching up thing is awful to work with.

**Comment by sspatari** [comment]
So guys, my problem was caused by Bracket Pair Colorized 2 vscode extention, after changing it to Bracket Pair Colorized ( simply version 1) or removing it integrally, vim extension was not lagging anymore.

**Comment by owenstake** [comment]
> Also, I think part of my problem was from the extension "Bracket Pairs Colorizer 2". The redrawing of colors every time the cursor moved made the lag worse. Uninstalling the extension helped, but did not solve the problem completely.
It helps! I close "Bracket Pairs Colorizer 2" and other paint extension to avoid repaint during Normal/Insert Mode change.

**Comment by Sheepolution** [comment]
I changed the color scheme to something else and back, and suddenly VS Code is fast again. Even after restarting it. Weird, but glad it's solved for me.

**Comment by pianocomposer321** [comment]
@Sheepolution what color scheme was it and what did you change it too? I think I remember having similar experiences, but I don't remember with which themes I had problems, and which ones were ok.

**Comment by Sheepolution** [comment]
@pianocomposer321 I used Argdown Dark, from the extension Argdown. But the weird thing is that if  I go back to this font now it's still fast!

**Comment by pianocomposer321** [comment]
@Sheepolution yeah that is strange. I'll be interested to see if it lasts.

**Comment by J-Fields** [comment]
@pianocomposer321 @Sheepolution Not sure what the cause of this is, but I'll throw this out there in case it's relevant: not all color schemes support semantic highlighting, and I think semantic highlighting on large projects probably comes with a performance cost (which is more noticeable when VSCodeVim is enabled).

**Comment by pianocomposer321** [comment]
@J-Fields Interesting thought. I doubt that's what was causing the lag for me though, b/c I have never done a project in VSCode that would be considered "large" (I switched fully to regular neovim a couple of months ago).

Unrelated, I have noticed a _lot_ more lag once the cursor gets past the point where the view has to scroll to keep the cursor visible. I don't think that it's just that the lag builds up and I don't notice it until that distance is achieved, partially because the lag doesn't happen nearly as much when using the down arrow rather than the j key.

**Comment by Sheepolution** [comment]
> I changed the color scheme to something else and back, and suddenly VS Code is fast again. Even after restarting it. Weird, but glad it's solved for me.

I might have been mistaken about this. I use VSCode Vim on both my own computer and the one from my job. I think what's going on is that VSCode was always faster at my job, but that I never noticed. So the color scheme changing thing might not even have made any differences.

Anyway, I was doing some experimenting and here's what I found.
Moving up and down is slow like I demonstrated in [my gif ](https://github.com/VSCodeVim/Vim/issues/3957#issuecomment-619360278) earlier. I fixed this by removing my remapping from j and k to gj and gk. This solves the issue of the document moving up even though I let go of the key. But it's still kind of slow. And I noticed that when in Normal Mode, moving up and down with the arrow keys is faster than with j and k. It's weird because I believe this is not the case on my work computer. I'm not sure why my own computer, which has better hardware, runs VS Code worse, or this extension at least. Perhaps some NVidia settings might solve the issue?

By the way, I tried to speed it up by disabling all other extensions, but that didn't work.

**Comment by Alraemon** [comment]
> Also, I think part of my problem was from the extension "Bracket Pairs Colorizer 2". The redrawing of colors every time the cursor moved made the lag worse. Uninstalling the extension helped, but did not solve the problem completely.

omg, thanks for your mention! this is exactly the extension causing issue

**Comment by lifeeric** [comment]
going back to the normal editor, giving up vim for now :(

**Comment by rahmnstein** [comment]
For me it is the dart plugin, which makes flutter development impossible with vim extension.

**Comment by mxs** [comment]
For me there is a huge pressing 'j' for cursor down in normal mode but not 'k'. I've tried uninstalling and reinstalling the extension to no avail 😢

**Comment by berknam** [comment]
> 
> 
> For me there is a huge pressing 'j' for cursor down in normal mode but not 'k'. I've tried uninstalling and reinstalling the extension to no avail 😢

Do you have `jk` or `jj` or any other mapping starting with 'j' on `normalModeKeyBindings` or `normalModeKeyBindingsNonRecursive`?

**Comment by mxs** [comment]
@berknam yes I do, `jj`. After removing `normalModeKeyBindings` I no longer experience this issue, also changed it to something like `pp` for the sake of it seems to fix this as well.

Now just removing `normalModeKeyBindingsNonRecursive`  did not resolve my issue.

This only become a problem very recently.

btw thank you @berknam!!!

**Comment by berknam** [comment]
@mxs This is normal Vim behavior. Previously the timeout feature wasn't fully implemented, now it is. This means that when you have a remap with more than one key starting with 'j' or 'k' in normal mode, when you press those keys it will wait for another key or for timeout to finish (1s by default). The same thing happens in real Vim.

**Comment by mxs** [comment]
@berknam thank for you highlighting that, I've only been a using vim full time with vscode.

However it is a bit curious that both vscode vim's [docs](https://github.com/VSCodeVim/Vim#key-remapping) have an example of `jj`. Also most the pages I've [found](https://vim.fandom.com/wiki/Avoid_the_escape_key) and [here](https://medium.com/diegodev/vim-tip-remapping-jj-to-escape-8efc4ad8a6bf) that have `jj` as the first example of a remap.

Ideally I would still like to use `jj`, so are there ideas how I might be able to do that without the lag in normal mode?

**Comment by berknam** [comment]
> 
> 
> @berknam thank for you highlighting that, I've only been a using vim full time with vscode.
> 
> However it is a bit curious that both vscode vim's [docs](https://github.com/VSCodeVim/Vim#key-remapping) have an example of `jj`. Also most the pages I've [found](https://vim.fandom.com/wiki/Avoid_the_escape_key) and [here](https://medium.com/diegodev/vim-tip-remapping-jj-to-escape-8efc4ad8a6bf) that have `jj` as the first example of a remap.
> 
> Ideally I would still like to use `jj`, so are there ideas how I might be able to do that without the lag in normal mode?

That remap is for insert mode not normal mode. If you want to use `jj` in insert mode to go back to normal mode you need to use this (on settings.json):
```jsonc
"vim.insertModeKeyBindings": [
    {
        "before": ["j", "j"],
        "after": ["<Esc>"]
    },
]
```

You previously had it in "normalModeKeyBindings" which was wrong.

**Comment by mxs** [comment]
@berknam awesome thats actually what I wanted! thank you!

**Comment by StitiFatah** [comment]
Same problem on Archlinux, after a while it takes around a second to switch between normal and insert mode and the input lag make it pretty unusable.  I used to think that was because I always had dozens of tabs so I switched to Vscode Neovim which doesn't have this problem, now that I'm back to VscodeVim I realized the same problem also happened with few tabs if I intensively use Vscode ( 9 hours at the moment ). 
I have the colorizing bar enabled + Bracket Pair Colorizer ( 1 not 2 ) installed. I'll uninstall it and deactivate the color switch and give you feedback. Did you guys still experience this "bug" ?

**Comment by leo-mao** [comment]
for those Haskell programmer, `Haskell-linter` cause this issue on win8.1 whilst Bracket Pairs Colorizer 2 is innocent.

**Comment by ncharchut** [comment]
Pylance was the issue for me (OSX 11.2) in Python files (shocker)

**Comment by amantiwari1** [comment]
@everyone
same here type is pretty slow any solution!!!

**Comment by joeytwiddle** [comment]
Do check out your plugins.

For example, I used to have this problem with `daosro.stylelint` but it went away when I switched to `stylelint.vscode-stylelint`

**Comment by Alex-EEE** [comment]
Bracket Pair Colorizer 2 here as well :-(. (mac Catalina)

Switched to Bracket Pair Colorizer (original version) and it's A OK

**Comment by Christopher-Hayes** [comment]
Happens to me every time I'm typing a Typescript type. I love having auto-complete, but this drives me nuts considering Vim commands briefly stop working all together.

**Comment by mikeRead** [comment]
Same here for me, and I cant bring my self to use this :.(

VIM extension disabled holding j (or any key)
![4ce96972a17624b2cc8639c4639b89e4](https://user-images.githubusercontent.com/3676392/122271639-4061a980-cea5-11eb-87f1-a4ffc234d7bd.gif)

With VIM enabled and in insert mode  holding j (or any key)
![3414040108f5bc87b2f4cbaf88cbbd20](https://user-images.githubusercontent.com/3676392/122271803-71da7500-cea5-11eb-9d60-215e19b6858f.gif)

Note: The same amount of "j"s will still be typed but when I release the key it just keeps going at this speed until all "j"s are typed.

**Comment by lassebenni** [comment]
For me the Auto Save "afterDelay" option seemed to cause the delay issues.

![image](https://user-images.githubusercontent.com/7990862/143588636-f2829ca2-ab08-4c02-81ea-fa9e258ea421.png)

I switched it off for now.

**Comment by rickyzhang82** [comment]
It is not only me who feel the same way. What is the fix/workaround?

It is unusable because I type more bugs than code.

**Comment by franchesoni** [comment]
Reading all "solutions" I suspect that the reason is that *the running time of another extension is causing a perceived lag in the vim extension* (which doesn't really explain why this happens at entering `insert` mode)

It happened to me with a large markdown file and the spell checker on

Does this make any sense?

**Comment by ZXYFrank** [comment]
SWITCH to VScode-Neovim :exclamation:
It works incredibly smoothly :exclamation:

**Comment by Mrcuve0** [comment]
Test this, it worked for me:
https://github.com/microsoft/vscode/issues/75627#issuecomment-1078827311

**Comment by RaviPabari** [comment]
> Test this, it worked for me: [microsoft/vscode#75627 (comment)](https://github.com/microsoft/vscode/issues/75627#issuecomment-1078827311)

Worked for me as well.

**Comment by awelzel** [comment]
Just leaving this as a datapoint - after disabling `git.autorefresh` of the `vscode.git` extension sluggishness improved noticeable for me (I believe it boils down to a large number of untracked files in the checkout I was working in).

![Screenshot-20220811-152240](https://user-images.githubusercontent.com/2100509/184143298-db7f045a-2430-40a8-99ee-6539c0eb083f.png)

**Comment by bariswheel** [comment]
Looking over the comments, disaapointing to see this is a problem for almost 4  years. Experiencing the same problems myself. Perhaps it's an extension conflict, would be nice to see if we all have a common extension installed, besides this one.

**Comment by Enside** [comment]
@J-Fields 

After several days of research around the problem, I suspect that vscode not send command directly to vim extension which cause latency and perhaps other redundant works. Then I add these keybindings to my `keybindings.json `:

```json
{
  {
    "args": {
      "after": ["a"]
    },
    "command": "vim.remap",
    "key": "a",
    "when": "editorTextFocus"
  },
  {
    "args": {
      "after": ["b"]
    },
    "command": "vim.remap",
    "key": "b",
    "when": "editorTextFocus"
  },
  {
    "args": {
      "after": ["c"]
    },
    "command": "vim.remap",
    "key": "c",
    "when": "editorTextFocus"
  },
  // ... from a to z. maybe from A to Z also needed.
}
```

Adding keybindings manually is tedious.  After all of these done, it seems that vscode send command to vim extension directly, navigation become smooth.

Can we just add these by vim extension automatically or find root cause of the problem through this clue.

**Comment by druskus20** [comment]
> @J-Fields
> 
> After several days of research around the problem, I suspect that vscode not send command directly to vim extension which cause latency and perhaps other redundant works. Then I add these keybindings to my `keybindings.json `:
> 
> ```json
> {
>   {
>     "args": {
>       "after": ["a"]
>     },
>     "command": "vim.remap",
>     "key": "a",
>     "when": "editorTextFocus"
>   },
>   {
>     "args": {
>       "after": ["b"]
>     },
>     "command": "vim.remap",
>     "key": "b",
>     "when": "editorTextFocus"
>   },
>   {
>     "args": {
>       "after": ["c"]
>     },
>     "command": "vim.remap",
>     "key": "c",
>     "when": "editorTextFocus"
>   },
>   // ... from a to z. maybe from A to Z also needed.
> }
> ```
> 
> Adding keybindings manually is tedious. After all of these done, it seems that vscode send command to vim extension directly, navigation become smooth.
> 
> Can we just add these by vim extension automatically or find root cause of the problem through this clue.

I removed all of my non-vim keybinds and I've only once had a slowdown desync/problem. Maybe that has something to do? 
* I must note that I'm usually running vscode without a lot of extensions active. I'd say... 5 to 10 probably.
* I am also running vscode on a 8c/16t CPU, both for my desktop and my laptop

**Comment by Enside** [comment]
@druskus20 Running with many extensions is one of the causes. I don't have extension development experience, but I guess that if there are many extension hosts communicating with vscode, it would make vim extension's commands being delayed to execute.

**Comment by cpakken** [comment]
I think VSCode VIM works on large files just fine ( > 1000 lines). its probably the typescript / javascript intellisense that is bogging it down.

Whenever I open a remote repository in VSCode, it opens into a 'trusted' workspace where a lot of the extensions are disabled or limited use. Even when the file in remote repository is > 1000 lines, vim runs like a hot knife through butter. I played around turning on and off some extensions and I highly suspect its Vscode's native typescript/javascript intellisense extension.

If I open a <200 line typescript file, but with heavy use of generics and complicated types, vim mode slows to a crawl. When vim is off, the normal editor works normally

🤷‍♂️

**Comment by onlurking** [comment]
For those who come latter, this affinity setting helped me out:
https://github.com/microsoft/vscode/issues/75627#issuecomment-1078827311

**Comment by Demianeen** [comment]
For those who had problems with lagging input from the keyboard in some files, cypress helper + Vim extension together were causing this problem for me.

**Comment by MarcosH911** [comment]
https://github.com/microsoft/vscode/issues/75627#issuecomment-1078827311 Fixes it, and if that doest fix it try disabling Github Copilot, Tabnine, Codeium or any similar extension

**Comment by jedwards1211** [comment]
@mikeRead does navigating up/down by holding `j`/`k` go faster with the neovim extension?
Some vim extensions unbind from the `type` command in edit mode, letting VSCode itself handle keyboard input in the main process instead of having to stupidly send all keypresses over IPC.  The VSCode team decided early on not to let any extensions run in the main process, which has had a disastrous impact on Vim emulation.  It was a truly awful design choice.

**Comment by jedwards1211** [comment]
@onlurking I've had affinity set for a long time and it doesn't fix the gradual slowdown.  I still suspect VSCode itself is at fault for this issue, but who knows.

**Comment by onlurking** [comment]
Hey @jedwards1211 I migrated to [vscode-neovim](https://github.com/vscode-neovim/vscode-neovim) quite few months ago, it has it fair share of issue (eg [sync issues](https://github.com/vscode-neovim/vscode-neovim/issues/2184) when using Supermaven/Copilot), but it doesn't have the same slow down problem of VSCodeVim.

**Comment by jhobz** [comment]
I noticed that the laggy scrolling with `j`/`k` would happen more when the "sticky scrolling" pane would appear at the top, such as navigating within nested methods or HTML tags. Hiding the sticky scrolling section (right-click inside the pane, uncheck the option) significantly improved the performance of vim for me, in case that helps anyone else! Annoying to need to do this, but it was getting to the point where I was restarting the window at least once an hour.

**Comment by RononDex** [comment]
This issue is most simply solved by just moving on to NeoVim 😆  (haven't used vscode in years)

**Comment by SpicyRicecaker** [comment]
@jhobz 

> I noticed that the laggy scrolling with `j`/`k` would happen more when the "sticky scrolling" pane would appear at the top, such as navigating within nested methods or HTML tags.

Yes, I observe this too! When the lag starts, it's very easy to trigger it by going under a visible sticky scroll scope. But I'm having trouble reproducing the lag from scratch, i.e. in the unlagged state after a VSCode restart, going under a visible sticky scroll scope doesn't cause lag. So frustrating because I like the sticky scroll feature but this has taken me out of flow several times when Vim becomes unusable T-T.

**Comment by v-rongxiao-microsoft** [comment]
> I noticed that the laggy scrolling with `j`/`k` would happen more when the "sticky scrolling" pane would appear at the top, such as navigating within nested methods or HTML tags. Hiding the sticky scrolling section (right-click inside the pane, uncheck the option) significantly improved the performance of vim for me, in case that helps anyone else! Annoying to need to do this, but it was getting to the point where I was restarting the window at least once an hour.

`"editor.stickyScroll.enabled": false`

**Comment by boschdrill** [comment]
> > I noticed that the laggy scrolling with `j`/`k` would happen more when the "sticky scrolling" pane would appear at the top, such as navigating within nested methods or HTML tags. Hiding the sticky scrolling section (right-click inside the pane, uncheck the option) significantly improved the performance of vim for me, in case that helps anyone else! Annoying to need to do this, but it was getting to the point where I was restarting the window at least once an hour.
> 
> `"editor.stickyScroll.enabled": false`

that workds for me, thanks
